### PR TITLE
vendor: buildkit v0.11.4-0.20230228113103-218e934edfba

### PR DESCRIPTION
**Pull request by vvoland**
changes: https://github.com/moby/buildkit/compare/v0.11.3...218e934edfba

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by crazy-max** [comment]
@vvoland Yes you can inline it in the meantime. This feature will be removed for next 0.12, see https://github.com/moby/buildkit/pull/3582.

See also https://github.com/moby/buildkit/blob/v0.11/docs/deprecated.md#build-information

File: builder/builder-next/exporter/writer.go [hunk 1]
```diff
@@ -19,6 +18,7 @@ import (
 // const (
 // 	emptyGZLayer = digest.Digest("sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1")
 // )
+const imageConfigField = "moby.buildkit.buildinfo.v1"
```
Comment (thaJeztah) [inline builder/builder-next/exporter/writer.go:21]: So, slightly wondering;

- As this is the master branch; does it already use the replacement? If so, perhaps we should (can we?) remove the functionality as a whole?
- If we _cannot_ remove it now, the alternative would be to add a `//nolint:...` (with a comment about the deprecation)

Advantage of the `//nolint:` would be that we deliberately make the build break once we would update BuildKit to v0.12, that way we don't accidentally forget about removing it once it's gone (whereas having a local `const` means we could still have (dead)code.

Comment (crazy-max) [reply builder/builder-next/exporter/writer.go:21]: Oh it's the master branch. Then yeah better to remove it entirely. related commit: https://github.com/moby/moby/pull/43239/commits/fda0226a890b9948a754c48fc3ad5f6e02f43f40

Comment (thaJeztah) [reply builder/builder-next/exporter/writer.go:21]: Thanks! Yes, so this is master (23.1.0 "dev"), which _will_ be the first release with BuildKit 0.11, but hasn't been released yet, so perhaps it's fine to deprecate/remove ahead of BuildKit.

So my suggestion would be;


- Open a PR to mark the feature deprecated (from the "Docker Engine" perspective) in https://github.com/docker/cli/blob/master/docs/deprecated.md. We can probably mark it as "deprecated in v23.0.0", and "to be removed in v23.1.0" (as it was already deprecated in BuildKit, but we didn't mention it)
- Remove it from master
- Vendor buildkit v0.11 "latest"

Comment (thaJeztah) [reply builder/builder-next/exporter/writer.go:21]: But I could use feedback on that suggestion, as I'm not very familiar with the feature (and who/how it's used) 😅

File: builder/builder-next/exporter/writer.go [hunk 2]
```diff
@@ -19,6 +18,7 @@ import (
 // const (
 // 	emptyGZLayer = digest.Digest("sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1")
 // )
```
[the remark this answers is not in the document]
Comment (vvoland) [reply builder/builder-next/exporter/writer.go:20]: Looks like no longer used. Last reference to this was removed in: 0728fb2b6395bd52524a1e5c134a1b37848a9c56
So probably okay, to remove?

**Comment by thaJeztah** [comment]
Flaky test (just commenting to allow us finding it back)

```
=== FAIL: github.com/docker/docker/pkg/plugins TestClientWithRequestTimeout (0.01s)
    client_test.go:254: assertion failed: expected an error, got nil: expected error
```

**Comment by thaJeztah** [comment]
This one can be rebased

**Comment by vvoland** [comment]
Rebased, are fine with sneaking the remove commented dead const in here, or should I make it a separate PR?

**Comment by thaJeztah** [comment]
I see you did in a separate commit, and it's a trivial changes, so seems fine to keep it here

**Review by thaJeztah — APPROVED** [review]
LGTM

(we're temporarily ahead of the last BuildKit release, but there likely will be a v0.11.4 with some bug fixes before we release)

**Comment by thaJeztah** [comment]
@rumpl @neersighted ptal 🤗

**Review by crazy-max — APPROVED** [review]

**Review by neersighted — APPROVED** [review]

**Pull request merged**